plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -54,6 +69,76 @@
 {
     private ElasticsearchQueryBuilder() {}
 
+    // Noticed that when running a sum() over values that are all null, ES returns 0.
+    // When the operation is not pushed down and is performed by Trino, the result is null.
+    // Incidentally, avg()/min()/max() has semantics that match Trino's (i.e. it returns null when run over a set of values that are all null).
+    // Therefore, when calculating SUM, stats aggregation is used here, so that the results of sum and avg can be obtained at the same time,
+    //  and then decide whether the result of sum should be corrected to null.
+    // See AggregationQueryPageSource#extractSumFromStatsValue
```

Comment:
I would suggest moving this comment right above the relevant line and perhaps making it a bit more concise. Perhaps something like:
```
// When running a sum() over values that are all null, ES returns 0, while Trino semantics 
// require the result to be null. Other aggregations, like avg()/min()/max() match Trino's 
// semantics, therefore, when calculating SUM, the stats aggregation is used, so that the 
// results of sum and min can be obtained at the same time, and then decide whether the 
// result of sum should be corrected to null.
```